test/functional/uri_options_spec.test.js

```diff
@@ -21,9 +21,12 @@ describe('URI Options (spec)', function() {
           metadata: { requires: { topology: 'single' } },
           test: function(done) {
             const query = qs.parse(url.parse(test.uri).query);
-            if (query.hasOwnProperty('tlsDisableOCSPEndpointCheck')) return this.skip();
-            if (query.hasOwnProperty('tlsDisableCertificateRevocationCheck')) return this.skip();
-
+            if (typeof query['tlsDisableOCSPEndpointCheck'] !== undefined) {
```

Comment:
Why change from `hasOwnProperty` here?

Reply:
also nit: prefer `query.property` to `query['property']` unless the property name includes escaped characters or spaces

Reply:
For some reason [repl.it here](https://repl.it/repls/DarkmagentaSecondGraph), when I changed from `qs`, to `querystring`, the returned object doesn't have `hasOwnProperty`.

```
require('querystring').parse('meow=true').hasOwnProperty('meow')
```

> TypeError: require(...).parse(...).hasOwnProperty is not a function

Reply:
@mbroadst reasoning here is that typescript doesn't allow to access unknown properties, so I'm more accustomed to do this above, plays better with TS. Will adopt your preference.

Reply:
> TypeError: require(...).parse(...).hasOwnProperty is not a function

Ah yeah, from the docs:

> The object returned by the querystring.parse() method does not prototypically inherit from the JavaScript Object. This means that typical Object methods such as obj.toString(), obj.hasOwnProperty(), and others are not defined and will not work.

It looks like you could do

`Object.prototype.hasOwnProperty.call(query, 'property');`

e.g. how [lodash.has](https://github.com/lodash/lodash/blob/master/has.js) does it.

Reply:
> reasoning here is that typescript doesn't allow to access unknown properties, so I'm more accustomed to do this above, plays better with TS. Will adopt your preference.

@nbbeeken and I discussed this [here](https://github.com/mongodb/node-mongodb-native/pull/2296#discussion_r408931122) - perhaps we should have a full group discussion about it? cc @mbroadst

Reply:
I agree with your proposal @emadum, but don't think we should hold up this PR any longer. It's just a few lines in a test file, we should continue the style discussion on @nbbeeken's PR and try to adhere to whatever decisions we make over there.